site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/model/Site.java

```diff
@@ -37,8 +39,26 @@ public Site(String name, String alias, String parentSiteId) {
         this.parentSiteId = parentSiteId;
     }
 
+
+    /**
+     * Site factory with name alias parent and ID
+     * make this a named factory to avoid mixing up site ID with parent site ID
+     * @param siteId site ID (never null)
+     * @param name the Site name
+     * @param alias the Site alias
+     * @param parentSiteId the Site parentSiteId
+     *
+     */
+    public static Site withIdNameAliasAndParent(final String siteId, String name, String alias, String parentSiteId) {
+        final Site site = new Site(name, alias, parentSiteId);
+        Objects.requireNonNull(siteId);
+        site.setSiteId(siteId);
+        return site;
+    }
+
```

Comment:
Currently we skip validating all kind of rules/checks for model since we assume they are pure with no logic.
If we want to include a factory probably best to name it accordingly and also have it in its own class?
Not sure about this one... but happy to be persuaded.

Reply:
It is in the way of a 'named constructor', so callers won't easily mix up the parent ID and the site ID. I can take out the null-check on site ID, or just replace it with a constructor, if you like.